### PR TITLE
Treat TiUP as a component when migrating

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

Treat TiUP as a component when migrating

### CODE REVIEW

File: tools/migrate/main.go [hunk 1]
```diff
@@ -316,10 +317,41 @@ func migrate(srcDir, dstDir string, rehash bool) error {
 		}
 	}
 
+	// Tread TiUP as a component
```
Comment (nrc) [inline tools/migrate/main.go:320]:
```suggestion
	// Treat TiUP as a component
```

File: tools/migrate/main.go [hunk 2]
```diff
@@ -316,10 +317,41 @@ func migrate(srcDir, dstDir string, rehash bool) error {
...
+	tiup := v0manifest.ComponentInfo{
+		Name: "tiup",
+		Desc: "Components manager for TiDB ecosystem",
```
Comment (nrc) [inline tools/migrate/main.go:323]:
```suggestion
		Desc: "Component manager for TiDB ecosystem",
```

File: tools/migrate/main.go [hunk 3]
```diff
@@ -316,10 +317,41 @@ func migrate(srcDir, dstDir string, rehash bool) error {
...
+		Versions: []v0manifest.VersionInfo{
+			{
+				Version: v0manifest.Version(tiupver.NewTiUPVersion().SemVer()),
+				Date:    time.Now().Format(time.RFC3339),
+				Entry:   "tiup",
+			},
+		},
+	}
+	for _, goos := range []string{"linux", "darwin"} {
+		for _, goarch := range []string{"amd64", "arm64"} {
```
Comment (nrc) [inline tools/migrate/main.go:336]: I think we don't need arm/OS X

Comment (lonng) [reply tools/migrate/main.go:336]: Yes, the darwin/arm64 won't be added to manifest because there is no tarball.

**Pull request merged**